cmd/nmi/main.go

```diff
@@ -58,6 +61,15 @@ func main() {
 		version.PrintVersionAndExit()
 	}
 
+	// check if the cni is kubenet from the --network-plugin defined in kubelet config
+	isKubenet, err := utils.IsKubenetCNI(*kubeletConfig)
+	if err != nil {
+		klog.Fatalf("failed to check if CNI plugin is kubenet, error: %+v", err)
+	}
+	if !*allowNetworkPluginKubenet && isKubenet {
+		klog.Fatalf("AAD Pod Identity is not supported for Kubenet. Review <url> for more details.")
```

Comment:
Let's update the docs so we can replace the `<url>` placeholder.

Reply:
Done!